### PR TITLE
stakepoold: context -> stakepool

### PR DESCRIPTION
**Pull request by JoeGruffins**
I do not think that context.go belongs in the rpc package.  If it were just relaying rpc commands, sure, but it does and will do more.  The name "context" also conflicts with context.Context, which we want to start using to pass shutdowns and "request-scoped values across API boundaries and between processes"

This pr moves context to its own package named stakepool, renames the main struct Stakepool, and renames all references to it stakepool.

this work is on top of #528

### CODE REVIEW

**Comment by JoeGruffins** [comment]
actually
AppContext -> Stakepoold
ctx -> spd

**Comment by JoeGruffins** [comment]
Thanks for thumbs up guys! I wasn't sure how this would be accepted.  I'll go ahead and rebase if #528 goes through then.

File: backend/stakepoold/stakepool/stakepool.go [hunk 1]
```diff
@@ -1,4 +1,8 @@
-package rpcserver
+// Copyright (c) 2019 The Decred developers
+
+//Package stakepool holds the Stakepool struct and processes incomming commands
```
Comment (JoeGruffins) [inline backend/stakepoold/stakepool/stakepool.go]: spacing

File: backend/stakepoold/signal.go [hunk 2]
```diff
@@ -19,13 +19,13 @@ var shutdownSignaled = make(chan struct{})
 // withShutdownCancel creates a copy of a context that is cancelled whenever
 // shutdown is invoked through an interrupt signal or from an JSON-RPC stop
 // request.
-func withShutdownCancel(ctx context.Context) context.Context {
-	ctx, cancel := context.WithCancel(ctx)
+func withShutdownCancel(spd context.Context) context.Context {
```
Comment (jholdstock) [inline backend/stakepoold/signal.go]: I think this should have remained `ctx context.Context`?

File: backend/stakepoold/server_test.go [hunk 3]
```diff
@@ -86,14 +86,14 @@ func randString(n int) string {
 }
 
 var (
-	c  *rpcserver.AppContext
-	wt rpcserver.WinningTicketsForBlock
+	c  *stakepool.Stakepoold
```
Comment (jholdstock) [inline backend/stakepoold/server_test.go]: Rename to `spd` for consistency?

File: backend/stakepoold/signal.go (hunk 2 above)
Comment (JoeGruffins) [reply backend/stakepoold/signal.go]: Good catch.

File: backend/stakepoold/server_test.go (hunk 3 above)
Comment (JoeGruffins) [reply backend/stakepoold/server_test.go]: Even better catch.

File: go.mod [hunk 4]
```diff
@@ -32,7 +32,6 @@ require (
 	github.com/gorilla/websocket v1.4.1 // indirect
 	github.com/jessevdk/go-flags v1.4.0
 	github.com/jrick/logrotate v1.0.0
-	github.com/kr/pretty v0.1.0 // indirect
```
Comment (JoeGruffins) [inline go.mod]: see, I switched to go 1.13 and It wants this change.

**Pull request merged**